cpp/tests/test_ide_secir.cpp

```diff
@@ -233,13 +245,79 @@ TEST(IdeSecir, checksimulationFunctions)
+
+    // add time points for initialization for transitions
+    Vec vec_init(num_transitions);
+    vec_init[(int)mio::isecir::InfectionTransition::SusceptibleToExposed]                 = 0.0;
+    vec_init[(int)mio::isecir::InfectionTransition::ExposedToInfectedNoSymptoms]          = 10.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedNoSymptomsToInfectedSymptoms] = 0.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedNoSymptomsToRecovered]        = 0.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedSymptomsToInfectedSevere]     = 10.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedSymptomsToRecovered]          = 0.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedSevereToInfectedCritical]     = 0.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedSevereToRecovered]            = 0.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedCriticalToDead]               = 0.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedCriticalToRecovered]          = 0.0;
+    // add initial time point to time series
+    init.add_time_point(-12, vec_init);
+    // add further time points
+    while (init.get_last_time() < 0) {
+        init.add_time_point(init.get_last_time() + dt, vec_init);
+    }
+
+    // Initialize model.
+    mio::isecir::Model model(std::move(init), N, Dead_before);
+
+    // Check that StateAgeFunctions are only considered equal if they are of the same derived
+    // class and have the same funcparam
+    mio::isecir::ExponentialDecay expdecay(0.5);
+    mio::isecir::ExponentialDecay expdecay2(0.5);
+    mio::isecir::ExponentialDecay expdecay3(1.0);
+    mio::isecir::SmootherCosine smoothcos(0.5);
+
+    EXPECT_TRUE(expdecay == expdecay2);
+    EXPECT_FALSE(expdecay == expdecay3);
+    EXPECT_FALSE(expdecay == smoothcos);
+
+    // Check that it also holds when a StateAgeFunctionWrapper is set with the respective functions
+    mio::isecir::StateAgeFunctionWrapper prob(expdecay);
+    mio::isecir::StateAgeFunctionWrapper prob2(expdecay2);
+    mio::isecir::StateAgeFunctionWrapper prob3(expdecay3);
+    mio::isecir::StateAgeFunctionWrapper prob4(smoothcos);
+
+    EXPECT_TRUE(prob == prob2);
+    EXPECT_FALSE(prob == prob3);
+    EXPECT_FALSE(prob == prob4);
+
+    // Check that it also holds when a parameter is set with the respective StateAgeFunctionWrapper
+    model.parameters.set<mio::isecir::TransmissionProbabilityOnContact>(prob2);
+    model.parameters.set<mio::isecir::RelativeTransmissionNoSymptoms>(prob3);
+    model.parameters.set<mio::isecir::RiskOfInfectionFromSymptomatic>(prob4);
+
+    EXPECT_TRUE(prob == model.parameters.get<mio::isecir::TransmissionProbabilityOnContact>());
+    EXPECT_FALSE(prob == model.parameters.get<mio::isecir::RelativeTransmissionNoSymptoms>());
+    EXPECT_FALSE(prob == model.parameters.get<mio::isecir::RiskOfInfectionFromSymptomatic>());
+}
```

Comment:
I think there are some new functions introduced in all the particular classes that have not yet been tested. Could you please do so? Could you also test that copy/clone works correctly and the the unique_ptrs are created correctly?

Reply:
I added some tests for the special member functions and setters and getters. But I don't know how to test if the unique_ptrs were created correctly.